### PR TITLE
Refine SOCKS5 and Tor implementation

### PR DESCRIPTION
**Pull request by bassosimone**
- Do not automatically use Tor's socks5
- Automatically use Tor instead when httpo://... URL are used

### CODE REVIEW

File: src/protocols/http.hpp [hunk 1]
```diff
@@ -558,6 +558,9 @@ class Request : public NonCopyable, public NonMovable {
         // Extend settings with address and port to connect to
         settings["port"] = serializer.port;
         settings["address"] = serializer.address;
+        if (serializer.schema == "httpo") {
+            settings["socks5_proxy"] = "127.0.0.1:9050";  // over Tor
+        }
```
Comment (hellais) [inline src/protocols/http.hpp]: This should not override the proxy settings that are specified by the user. That is it should be possible to specify an alternate Tor SOCKS port and make sure such setting has priority over defaulting to 9050.

Comment (bassosimone) [inline src/protocols/http.hpp]: Agreed, see fd17a53b6...

**Comment by bassosimone** [comment]
Hi @hellais,

I've created a setting called `tor_socks_port` that takes precedence over `socks5_proxy` and is used only when `httpo://...` URLs are used.

While there, I've also changed the code not to modify the settings object, but to make changes on a private copy of it, because it's surprising that you pass a settings object to a function and after that function returns the object is unchanged.

What do you think?

**Comment by bassosimone** [comment]
Ok, with the previous commit I've improved the regress tests so that they do not fail and it's obvious that the behavior with respect to `tor_socks_port` and `socks5_proxy` is correct.

**Comment by hellais** [comment]
Beautiful! Thanks!

**Pull request merged**